### PR TITLE
Add snap to grid feature

### PR DESCRIPTION
**Pull request by comet0**
Adds an invisible grid with 'snap' points every 75px.
Can be enabled via checkbox in View menu.

No idea how to create input boxes to accept user inputted grid sizes.

### CODE REVIEW

File: gns3/items/node_item.py [hunk 1]
```diff
@@ -377,6 +377,10 @@ def itemChange(self, change, value):
         :param value: value of the change
         """
 
+        if change == 9 and self._main_window.uiSnapToGridAction.isChecked(): # Unsure but change seems to alternate between 0/9 while dragging
```
Comment (noplay) [inline gns3/items/node_item.py]: This documentation will help you to know the name of the event:
http://doc.qt.io/qt-5/qgraphicsitem.html

In your case it's QtWidgets.QGraphicsItem.ItemPositionHasChanged

**Comment by noplay** [comment]
Great I will help you to make the PR mergable

**Comment by noplay** [comment]
gns3/ui/main_window_ui.py is generated from gns3/ui/main_window.ui

You need to modify gns3/ui/main_window.ui with Qt Designer (or via the XML by hand but QtDesigner is recommended) and after that run:
python scripts/build_pyqt.py

This will regenerate the Python file.

**Comment by noplay** [comment]
I could make the modifications myself but if you agree I will help you to jump in the code base.

**Comment by comet0** [comment]
Should I commit the changed main_window_ui.py after build_pyqt re-creates it?
Seems like a lot of unnecessary changes: https://gist.github.com/comet0/47d07fc44c4635b57b4baed3091457cd

**Comment by noplay** [comment]
Yes you should commit it

**Comment by noplay** [comment]
Work nice just tested it :)

**Comment by noplay** [comment]
Perfect I will merge it and we will include it in 1.5.

Can you sign our contributor licence agreement?
https://github.com/GNS3/cla

It's require by our lawyers to avoid legal trouble 

What name do you want to me add to the Thanks to dialog?

**Comment by noplay** [comment]
Merged in 1.5

**Pull request closed without merging**